lib/process-services-cloud/src/lib/form/components/widgets/data-table/data-table-adapter.widget.spec.ts

```diff
@@ -0,0 +1,76 @@
+/*!
+ * @license
+ * Copyright © 2005-2023 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { WidgetDataTableAdapter } from './data-table-adapter.widget';
+import {
+    mockCountriesData,
+    mockCountriesIncorrectData,
+    mockInvalidSchemaDefinition,
+    mockSchemaDefinition,
+    mockSchemaDefinitionWithDifferentTypes
+} from '../../../mocks/data-table-widget.mock';
+import { ObjectDataRow } from '@alfresco/adf-core';
+
+describe('WidgetDataTableAdapter', () => {
+    let widgetDataTableAdapter: WidgetDataTableAdapter;
+
+    beforeEach(() => {
+        widgetDataTableAdapter = new WidgetDataTableAdapter(mockCountriesData, mockSchemaDefinition);
+    });
+
+    it('should set columns type to "text" during initialization', () => {
+        widgetDataTableAdapter = new WidgetDataTableAdapter(mockCountriesData, mockSchemaDefinitionWithDifferentTypes);
+
+        widgetDataTableAdapter.getColumns().forEach(column =>
+            expect(column.type).toBe('text')
+        );
+    });
+
+    it('should return rows if all columns are linked to data', () => {
+        const rows = widgetDataTableAdapter.getRows();
+
+        expect(rows).toEqual([
+            new ObjectDataRow({ id: 'IT', name: 'Italy' }),
+            new ObjectDataRow({ id: 'PL', name: 'Poland' }),
+            new ObjectDataRow({ id: 'UK', name: 'United Kingdom' })
+        ]);
+    });
+
+    it('should return an empty array if not all columns are linked to data', () => {
```

Comment:
if no columns are configured in the data source?

Reply:
we can improve description, but this case is not about data source that contains no columns, its about
data for **some** (one or more) column is missing
Example:
column schema:
```
[
    {
        type: 'text',
        key: 'id',
        title: 'Country ID',
        sortable: true,
        draggable: true
    },
    {
        type: 'text',
        key: 'name',
        title: 'Country Name',
        sortable: true,
        draggable: true
    }
]
```

has two columns, id and name.
[requirement to render data table ](https://alfresco.atlassian.net/browse/AAE-15815?filter=-1)is data source must contain data linked to all columns,
so for example this data(rows) source:
```
[
    {
        id: 'PL'
    },
    {
        id: 'IT'
    }
]
```
is linked only to **id** column, and **name** column has no data to display